src/rust/engine/src/nodes.rs

```diff
@@ -303,14 +306,31 @@ impl Select {
           env_var_parts[2 * i + 1].clone(),
         );
       }
+
+      // TODO: Make this much less unwrap-happy as part of moving this to run off-thread and
+      // asynchronously.
+
```

Comment:
There might also be a potential solution involving interning `Snapshot` objects by digest on the rust side, and then keeping the python object more svelte? ... depends on the Snapshot manipulation API more than the the threading, per-se.

It might be worth opening a Snapshot manipulation API ticket soon and then doing some design sketching there.

Reply:
Filed // TODO: Make this much less unwrap-happy with https://github.com/pantsbuild/pants/issues/5502 and linked to it there